Lib/test/test_concurrent_futures.py

```diff
@@ -964,6 +970,37 @@ def test_ressources_gced_in_workers(self):
         mgr.shutdown()
         mgr.join()
 
+    def test_saturation(self):
+        executor = self.executor_type(4)
+        def acquire_lock(lock):
+            lock.aquire()
+
+        mp_context = get_context()
+        sem = mp_context.Semaphore(0)
+        job_count = 15 * executor._max_workers
+        for _ in range(job_count):
+            executor.submit(acquire_lock, sem)
+        self.assertEqual(len(executor._processes), executor._max_workers)
+        for _ in range(job_count):
+            sem.release()
+        executor.shutdown()
```

Comment:
Should use `try..finally` or, better, add `self.addCleanup(executor.shutdown)` at the beginning of this method.

Reply:
Yeah, I would prefer `self.addCleanup(executor.shutdown)`. IMO, it helps to demonstrate that the `executor.shutdown()` isn't really part of the above test.

Reply:
Actually, I just realized something: since we can use `threading.Semaphore()`, we can simply move the existing `test_saturation` and `test_idle_thread_reuse` from `ThreadPoolExecutorTest` to `ExecutorTest`, instead of having basically the same test replicated in both `ThreadPoolExecutorTest` and `ProcessPoolExecutorTest` (including the suggested changes). I'll rename `test_idle_thread_reuse` -> `test_idle_worker_reuse`.